_posts/2020-01-10-ios-logout-button-at-last.markdown

```diff
@@ -0,0 +1,93 @@
+from our API, and our code assumed that these values would never change. This is further complicated by the fact
+that our app is split into two pieces: the native code (written in Objective-C and Swift) and the React Native code
+(written in TypeScript). You can effectively think of these as _two_ apps that interoperate with each other. The
+state now has to be managed across _two_ pieces of software, further complexifying the work to add a log out
+button.
+
+Returning to the idea of developing features in mind from the start (versus adding them after the fact), our React
+Native codebase had always assumed a logged-in user. Adding support for this later on was too difficult (indeed,
+our solution was to
+[invalidate the React Native runtime upon log out](https://github.com/artsy/emission/pull/2027/files#diff-0cc174f9197fd0b06ecbd2eaa0247833R1020)).
+This wasn't a limitation of React Native, but rather it was a limitation of how we chose to organize our code. If
+we'd added a log out button earlier in the product lifetime, it wouldn't have been so difficult. The longer we
+waited, the more and more code we wrote that implicitly relied upon our existing limitations.
+
+In this way, the _absence_ of a feature had gained its own inertia. The missing feature became a present absence,
+and I think there's more to think about there – maybe for another blog post.
+
+So what we had was a difficult technical problem that wasn't _that_ high of priority and didn't have an obvious
+solution. Our product team wasn't feeling the pain, but our colleagues who interface more directly with our users
+_were_ feeling it.
+
+Changing gears a bit, Artsy kicked off 2020 with a company-wide Hackathon. We run these events roughly once year,
+and they provide a great opportunity for engineers and non-engineers to work together to build something over a few
+days. Maybe it's a brand-new piece of software to help us do our jobs better. Maybe it's a new zine to collect our
+favourite artworks from the site? And maybe, just maybe, it's a long-neglected user feature that never made the cut
+in prioritization meetings.
+
+Among all the ideas that Artsy staff submitted for the Hackathon, the Artsy iOS Log Out button received the third
+most votes.
+
+![Screenshot of our Hackathon ideas board](/images/2020-01-14-ios-logout-button-at-last/idea.png)
+
+I sat down with another Mobile Experience engineer, Brian, and the people who submitted the Hackathon idea. If we
+were going to build this, it was worth doing right, so we asked questions and learned more about why users need a
+log out button at all. I'll spare you the details, but it suffices to say that I learned a lot.
+
+Brian and I worked on the feature, digging into the internals of our app and the interop between native and React
+Native code. We learned a tonne about the React Native bridge, the existing architecture of our app, and how we'd
+like to see that mature going forward.
```

Comment:
This feels a bit [rest of the fucking owl](https://knowyourmeme.com/memes/how-to-draw-an-owl) to me 😉 - because this post is framed as being about the login button, I expect to get a little more into the technical details.

In the intro to this post, you mostly focus on the log-out button itself. You mention the hackathon right before the intro ends, but that's about it. The conclusion of the post, however, focuses mostly on the hackathon - I feel like there's a bit of a contractural mismatch, if you will. Somewhere in the middle of the post, this became more about _what_ was happening as opposed to _how_ it was getting done.

Since it seems like this post is more focused on the hackathon and on the difficulty of integrating new features than on the technical details of implementing the button, I would try to make that more clear in the title, the intro, etc. How could this post really consistently communicate its intent and content from the first impression on down?